discovery_engine_core/core/src/engine.rs

```diff
@@ -641,6 +641,7 @@ async fn update_stacks<'a>(
     // Separate stack documents (via the `stack` parameter) are not needed here, since they are
     // already contained in `all_documents`.
     all_documents = DuplicateFilter::apply(history, &[], all_documents);
+    all_documents = filter_semantically(all_documents, &SemanticFilterConfig::default());
```

Comment:
❓ should we group those filters into a common filter struct? similarly as we do with the `CommonFilter` for the per-stack filtering? not sure what the best name for this would be, maybe `GlobalCommonFilter` or sth similar?

Reply:
If you ask me, it's fine like this for the time being. The cross-stack filtering is relatively new, and we don't yet know where it's headed. It's already not a unified interface, because the different filters require different parameters. I would  see what changes are in store in the next month or two, and if we see that we added / are about to add a bunch of functionality here, then we can still refactor it to a struct / function. wdyt?